python/pyspark/ml/param/__init__.py

```diff
@@ -170,6 +170,15 @@ def toVector(value):
         raise TypeError("Could not convert %s to vector" % value)
 
     @staticmethod
+    def toMatrix(value):
+        """
+        Convert a value to ML Matrix, if possible
```

Comment:
While I am aware of this, distinction between `ml.linalg` and `mllib.linalg`, is a common source of confusion for the PySpark users. Of course we could be more forgiving, and automatically convert objects to the required class.